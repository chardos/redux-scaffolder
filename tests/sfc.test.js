const rsk = require('../src/index.js');
const fs = require('fs-extra');
const makeDir = require('make-dir');
const del = require('del');
const tmpPath = `${process.cwd()}/tmp`;

describe('When running rsk in the command line', () => {
  beforeEach(async () => {
    await makeDir('tmp');
    process.argv.splice(2);
  });

  afterEach(async () => {
    await del(`${tmpPath}/*`);
  })

  it('stateless functional component should match the snapshot', async () => {
    process.argv.push('sfc', 'potato', '--codeDirectory=tmp');
    await rsk();

    const output = await fs.readFile(`${tmpPath}/components/Potato/index.js`);
    expect(output.toString()).toMatchSnapshot()
  });
})