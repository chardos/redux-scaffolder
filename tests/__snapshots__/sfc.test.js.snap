// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When running rsk in the command line stateless functional component should match the snapshot 1`] = `
"import React from \\"react\\";

const Potato = () => {
  return <div>Potato</div>;
};

export default Potato;
"
`;
